test/suite/test_debug_mode05.py

```diff
@@ -0,0 +1,86 @@
+# means.
+#
+# In jurisdictions that recognize copyright laws, the author or authors
+# of this software dedicate any and all copyright interest in the
+# software to the public domain. We make this dedication for the benefit
+# of the public at large and to the detriment of our heirs and
+# successors. We intend this dedication to be an overt act of
+# relinquishment in perpetuity of all present and future rights to this
+# software under copyright law.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,
+# EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF
+# MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.
+# IN NO EVENT SHALL THE AUTHORS BE LIABLE FOR ANY CLAIM, DAMAGES OR
+# OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE,
+# ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR
+# OTHER DEALINGS IN THE SOFTWARE.
+
+import wiredtiger, wttest
+
+def timestamp_str(t):
+    return '%x' %t
+
+# test_debug_mode05.py
+#     As per WT-5046, the debug table logging settings prevent rollback to
+#     stable in the presence of prepared transactions.
+#
+#     This test is to confirm the fix and prevent similar regressions.
+class test_debug_mode05(wttest.WiredTigerTestCase):
+    conn_config = 'log=(enabled),debug_mode=(table_logging=true)'
+    session_config = 'isolation=snapshot'
+    uri = 'file:test_debug_mode05'
+
+    def test_table_logging_rollback_to_stable(self):
+        self.session.create(self.uri, 'key_format=i,value_format=u')
+        cursor = self.session.open_cursor(self.uri, None)
+
+        self.conn.set_timestamp('stable_timestamp=' + timestamp_str(100))
+
+        # Try doing a normal prepared txn and then rollback to stable.
+        self.session.begin_transaction()
+        for i in range(1, 50):
+            cursor[i] = b'a' * 100
+        self.session.prepare_transaction(
+            'prepare_timestamp=' + timestamp_str(150))
+        self.session.timestamp_transaction(
+            'commit_timestamp=' + timestamp_str(200))
+        self.session.timestamp_transaction(
+            'durable_timestamp=' + timestamp_str(250))
+        self.session.commit_transaction()
+
+        self.conn.rollback_to_stable()
+
+        # The original bug happened when we had a txn that:
+        # 1). Was prepared.
+        # 2). Did not cause anything to be written to the log before committing.
+        # 3). Was the last txn before the rollback to stable call.
+        # Therefore, we're specifically not doing any operations here.
+        self.session.begin_transaction()
+        self.session.prepare_transaction(
+            'prepare_timestamp=' + timestamp_str(300))
+        self.session.timestamp_transaction(
+            'commit_timestamp=' + timestamp_str(350))
+        self.session.timestamp_transaction(
+            'durable_timestamp=' + timestamp_str(400))
+        self.session.commit_transaction()
+
```

Comment:
It would be good to add a comment here about what happens during this call when the bug existed versus what we expect now. I.e. the nature of the original bug. Did the call hang? Take an assertion failure? Return incorrect data?

Reply:
cf2c797